### PR TITLE
Plugins: Add option to disable TLS in the socks proxy

### PR DESCRIPTION
**Pull request by PoorlyDefinedBehaviour**
**What is this feature?**

Adds `allow_insecure: bool` to `secure_socks_datasource_proxy` config section. The socks proxy won't use TLS when `allow_insecure` is set to `true`. 

**Why do we need this feature?**

From the issue description:
> Sometimes there's a use case for testing grafana's SOCKs backend locally, and this would be significantly easier if we could disable the TLS requirement for the socks backend.

**Who is this feature for?**

Plugin developers

**Which issue(s) does this PR fix?**:

Part of https://github.com/grafana/hosted-grafana/issues/4935

Depends on https://github.com/grafana/grafana-plugin-sdk-go/pull/833

**TODO**:  
- [x] Update the plugin sdk dependency to the correct version before merging

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -475,6 +474,8 @@ require (
 	go.opentelemetry.io/proto/otlp v1.0.0 // indirect
 )
 
+require github.com/grafana/grafana-plugin-sdk-go v0.196.1-0.20231205151856-c5479f051888
```
Comment (PoorlyDefinedBehaviour) [inline go.mod:477]: Using a specific commit from my [PR](https://github.com/grafana/grafana-plugin-sdk-go/pull/833) in the sdk repo. Will update to the correct sdk version after the PR in the sdk repo is merged.

**Review by marefr — APPROVED** [review]
Great LGTM

Sidenote: It's convenient to use a replace directive in go.mod when temporary working with dependency update. You can use `go mod edit -replace=github.com/grafana/grafana-plugin-sdk-go=github.com/grafana/grafana-plugin-sdk-go@<branch name or commit>`.

**Review by chri2547 — APPROVED** [review]

File: docs/sources/setup-grafana/configure-grafana/proxy/index.md [hunk 2]
```diff
@@ -32,14 +32,15 @@ To complete this task, you must first deploy a socks proxy server that supports
...
-   | `proxy_address` | the address of the proxy                   | localhost:9090                  |
+   | Key              | Description                                | Example                         |
+   | ---------------- | ------------------------------------------ | ------------------------------- |
+   | `enabled`        | Enable this feature in Grafana             | true                            |
+   | `root_ca_cert`   | The file path of the root ca cert          | /etc/ca.crt                     |
+   | `client_key`     | The file path of the client private key    | /etc/client.key                 |
+   | `client_cert`    | The file path of the client public key     | /etc/client.crt                 |
+   | `server_name`    | The domain name of the proxy, used for SNI | proxy.grafana.svc.cluster.local |
+   | `proxy_address`  | The address of the proxy                   | localhost:9090                  |
+   | `allow_insecure` | Disable TLS in the socks proxy             | localhost:9090                  |
```
Comment (dafydd-t) [inline docs/sources/setup-grafana/configure-grafana/proxy/index.md:43]: I think we need to change the example value in the new line to be a boolean.

Comment (PoorlyDefinedBehaviour) [reply docs/sources/setup-grafana/configure-grafana/proxy/index.md]: Fixed. It was a copy pasting mistake

File: go.mod (hunk 1 above)
Comment (PoorlyDefinedBehaviour) [reply go.mod]: Updated to the sdk version [v0.197.0](https://github.com/grafana/grafana-plugin-sdk-go/releases/tag/v0.197.0)

**Pull request merged**